### ISSUE TITLE
Updates to cloudflare_record resources don't work when using deprecated value field

### ISSUE THREAD

**Issue by zackpollard** [cloudflare/terraform-provider-cloudflare#3890]
### Confirmation

- [X] This is a bug with an existing resource and is not a feature request or enhancement. Feature requests should be submitted with Cloudflare Support or your account team.
- [X] I have searched the issue tracker and my issue isn't already found.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

4.39.0 and 4.40.0 both exhibit this behaviour

### Affected resource(s)

cloudflare_record

### Terraform configuration files

```hcl
resource "cloudflare_record" "immich_app_release_domain" {
  name = "a.update.test.example.org"
  proxied = true
  ttl = 1
  type = "CNAME"
  value = "test.example.org"
  zone_id = var.zone_id
}
```

### Link to debug output

https://gist.github.com/zackpollard/85c3a4922e9d5221b12dcbd61749a147

### Panic output

No crash

### Expected output

The dns record to change in cloudflare.

### Actual output

The dns record value remains unchanged, terraform completes successfully and without manual inspection you wouldn't know what had happened.

### Steps to reproduce

1. Create resource given above
2. Alter resource value
3. Apply changes
4. Apply changes again and see that terraform tries to update it for a second time
5. Check dashboard, record is not updated.

### Additional factoids

It looks like this may be fixed in main, but not yet released (link included in references). I would like to request a new provider is released ASAP. The current behavior where the value field that was the recommended method up to two releases ago, silently fails to update, is very scary. I'm surprised a bugfix version wasn't immediately released, this could be breaking production deployments all over the place without people realising what is going on.

### References

https://github.com/cloudflare/terraform-provider-cloudflare/commit/e4e87a2f2601b7eb66fd96191744aa231eca0c6d

**Comment by zackpollard** [comment]
Sorry, just noticed the debug logs are from 4.39.0 as I was testing which previous versions were also affected. If you would like me to attach logs from 4.40.0 I can do that, but the bug absolutely still exists in that version.

**Comment by jacobbednarz** [comment]
this is already fixed by #3776 and will be released with the next mainline release based on https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/contributing/release-process.md

if you want to use the fix immediately, you can build a custom version following the docs at https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/contributing/using-non-released-versions.md

**Comment by zackpollard** [comment]
> this is already fixed by #3776 and will be released with the next mainline release based on https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/contributing/release-process.md
> 
> if you want to use the fix immediately, you can build a custom version following the docs at https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/contributing/using-non-released-versions.md

Right, I'm just surprised this isn't deemed as needing a hotfix. The two latest versions out there right now won't update DNS records values if you were using the previously accepted `value` property. That seems like a pretty serious bug and the worst part is terraform succeeds and says the resource was updated.
